sdk/nodejs/config.ts

```diff
@@ -49,13 +49,14 @@ export class Config {
      * @param key The key to lookup.
      * @param opts An options bag to constrain legal values.
      */
-    public get(key: string, opts?: StringConfigOptions): string | undefined {
-        const v = getConfig(this.fullKey(key));
+    public get<K extends string = string>(key: string, opts?: StringConfigOptions<K>): K | undefined {
+        const v: string | undefined = getConfig(this.fullKey(key));
         if (v === undefined) {
             return undefined;
         }
         if (opts) {
-            if (opts.allowedValues !== undefined && opts.allowedValues.indexOf(v) === -1) {
+            // SAFETY: if allowedValues != null, verifying v ∈ K[]
```

Comment:
i'm not sure that this comment adds any value.  it seems to just be restating the code below.

Reply:
I tend to write `SAFETY` comments when performing potentially unsafe casts (`v as K`). Can remove.

Reply:
I'm ok with that.  We can take this as is. :)